src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/VilkårperiodeService.kt

```diff
@@ -79,22 +87,61 @@ class VilkårperiodeService(
         }
     }
 
-    private fun opprettGrunnlagsdata(behandlingId: UUID) = vilkårperioderGrunnlagRepository.insert(
-        VilkårperioderGrunnlagDomain(
-            behandlingId = behandlingId,
-            grunnlag = VilkårperioderGrunnlag(
-                aktivitet = GrunnlagAktivitet(
-                    aktivitetService.hentAktiviteterForGrunnlagsdata(
-                        behandlingService.hentSaksbehandling(behandlingId).fagsakPersonId,
-                        fom = LocalDate.now().minusMonths(3),
-                        tom = LocalDate.now().plusYears(1),
-                    ),
-                    tidspunktHentet = LocalDateTime.now(),
-                ),
+    private fun opprettGrunnlagsdata(behandlingId: UUID): VilkårperioderGrunnlagDomain {
+        val søknad = søknadService.hentSøknadBarnetilsyn(behandlingId)
+        val utgangspunktDato = søknad?.mottattTidspunkt ?: LocalDateTime.now()
+
+        val fom = YearMonth.from(utgangspunktDato).minusMonths(3).atDay(1)
+        val tom = YearMonth.from(utgangspunktDato).plusYears(1).atEndOfMonth()
+
+        val grunnlag = VilkårperioderGrunnlag(
+            aktivitet = hentGrunnlagAktvititet(behandlingId, fom, tom),
+            ytelse = hentGrunnlagYtelse(behandlingId, fom, tom),
+            hentetInformasjon = HentetInformasjon(
+                fom = fom,
+                tom = tom,
+                tidspunktHentet = LocalDateTime.now(),
+            ),
+        )
+
+        return vilkårperioderGrunnlagRepository.insert(
+            VilkårperioderGrunnlagDomain(
+                behandlingId = behandlingId,
+                grunnlag = grunnlag,
             ),
+        )
+    }
+
+    private fun hentGrunnlagAktvititet(
+        behandlingId: UUID,
+        fom: LocalDate,
+        tom: LocalDate,
+    ) = GrunnlagAktivitet(
+        aktiviteter = aktivitetService.hentAktiviteterForGrunnlagsdata(
+            ident = behandlingService.hentSaksbehandling(behandlingId).ident,
+            fom = fom,
+            tom = tom,
         ),
     )
 
+    private fun hentGrunnlagYtelse(
+        behandlingId: UUID,
+        fom: LocalDate,
+        tom: LocalDate,
+    ): GrunnlagYtelse {
+        val ytelserFraRegister = ytelseService.hentYtelseForGrunnlag(behandlingId, fom, tom)
```

Comment:
Når man har flere argument så kan det være fint å bruke `named arguments`
```suggestion
        val ytelserFraRegister = ytelseService.hentYtelseForGrunnlag(behandlingId, fom = fom, tom = tom)
```